charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -69,11 +69,46 @@ spec:
           value: "/consul/login/acl-token"
         {{- end }}
         - name: CONSUL_HTTP_ADDR
+          {{- if .Values.client.enabled }}
+          {{/*
+            We use client agent nodes if we have them to support backwards compatibility in <=0.4 releases which
+            require connectivity between the registered Consul agent node and a deployment for health checking
+            (originating from the Consul node). Always leveraging the agents in the case that they're explicitly
+            opted into allows us to support users with agent node + "externalServers" configuration upgrading a
+            helm chart without upgrading api gateways. Otherwise, using "externalServers" when provided
+            without local agents will break gateways <=0.4.
+          */}}
           {{- if .Values.global.tls.enabled }}
-          value: https://$(HOST_IP):8501
+          value: $(HOST_IP):8501
           {{- else }}
-          value: http://$(HOST_IP):8500
+          value: $(HOST_IP):8500
+          {{- end }}
+          {{- else if .Values.externalServers.enabled }}
+          {{/*
+            "externalServers" specified and running in "agentless" mode, this will only work 0.5+
```

Comment:
```suggestion
            "externalServers" specified and running in "agentless" mode, this will only work with
            Consul API Gateway v0.5 or newer
```